javascript/index.js

```diff
@@ -24,17 +24,85 @@
     document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
     document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
   }, (error) => console.log(error));
-
+*/
 
 
 // Iteration 1 - using callbacks
-// ...
+getInstruction("mashedPotatoes", 0, (step1) => {
+  document.querySelector("#mashedPotatoes").innerHTML += `<li>${step1}</li>`;
+  getInstruction("mashedPotatoes", 1, (step2) => {
+    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step2}</li>`;
+    getInstruction("mashedPotatoes", 2, (step3) => {
+      document.querySelector("#mashedPotatoes").innerHTML += `<li>${step3}</li>`;
+      getInstruction("mashedPotatoes", 3, (step4) => {
+        document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
+        getInstruction("mashedPotatoes", 4, (step5) => {
+          document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
+          document.querySelector("#mashedPotatoes").innerHTML += `<li>Mashed potatoes are ready!</li>`;
+          document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
+        }, (error) => console.log(error));
+      }, (error) => console.log(error));
+    }, (error) => console.log(error));
+  }, (error) => console.log(error));
+}, (error) => console.log(error));
 
 // Iteration 2 - using promises
-// ...
+obtainInstruction('steak', 0)
+  .then( (step1) => document.querySelector("#steak").innerHTML += `<li>${step1}</li>`)
+  .then( () => obtainInstruction('steak', 1))
 
-// Iteration 3 using async/await
-// ...
+  .then( (step2) => {
+    document.querySelector("#steak").innerHTML += `<li>${step2}</li>`;
+  })
+  .then( () => obtainInstruction('steak', 2))
+  .then( (step3) => {
+    document.querySelector("#steak").innerHTML += `<li>${step3}</li>`
+    
+  })
+  .then( () => obtainInstruction('steak', 3))
+
+  .then( (step4) => {
+    document.querySelector("#steak").innerHTML += `<li>${step4}</li>`
+   
+  })
+  .then( () => obtainInstruction('steak', 4))
+  .then( (step5) => {
+    document.querySelector("#steak").innerHTML += `<li>${step5}</li>`
+    
+  })
+  .then( () => obtainInstruction('steak', 5))
+  .then( (step6) => {
+    document.querySelector("#steak").innerHTML += `<li>${step6}</li>`
+    
+  })
+  .then( () => obtainInstruction('steak', 6))
+  .then( (step7) => {
+    document.querySelector("#steak").innerHTML += `<li>${step7}</li>`
+  })
+  .then( () => obtainInstruction('steak', 7))
+  .then( () =>document.querySelector("#steak").innerHTML += `<li>Stake is ready!</li>`)
 
+// Iteration 3 using async/await
+async function makeBroccoli() {
+  try {
+    let step1=await obtainInstruction(0); 
+    document.querySelector("#broccoli").innerHTML += `<li>${step1}</li>`;
+    let step2=await obtainInstruction(1);
+    document.querySelector("#broccoli").innerHTML += `<li>${step2}</li>`;
```

Comment:
you could have stored the result of `document.querySelector("#broccoli")` in a variable. It could have saved you some lines ans should still work the same way